marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/CallBatcherImpl.java

```diff
@@ -575,7 +610,12 @@ private void retry(CallManager.CallEvent event, boolean callFailListeners) {
 
         @Override
         public CallBatcher<Void, E> forArgsGenerator(CallArgsGenerator<E> generator) {
-            return new CallBatcherImpl(client, caller, Void.class, generator);
+            return new CallBatcherImpl(client, caller, Void.class, generator).finishConstruction();
+        }
+
+        @Override
+        public CallBatcher<Void, E> forArgsGenerator(CallArgsGenerator<E> generator, String forestName) {
+            return new CallBatcherImpl(client, caller, Void.class, generator, forestName);
```

Comment:
For consistency, should we move the finishConstruction() call out of the constructor on line 110 and chain the call here on line 618?